server/src/test/java/io/crate/types/GeoShapeTypeTest.java

```diff
@@ -216,8 +216,17 @@ public void test_bytes_estimate() throws Exception {
 
     @Test
     public void test_check_equality_with_itself() throws Exception {
+        UnaryOperator<Map<String, Object>> normalizeSelect = geoJson -> type.implicitCast(geoJson);
+        UnaryOperator<Map<String, Object>> normalizeInsert = geoJson -> type.sanitizeValue(geoJson);
+
         for (Map<String, Object> json: geoJson) {
             assertThat(type.compare(json, json)).isEqualTo(0);
+
+            assertThat(type.compare(normalizeSelect.apply(json), json)).isEqualTo(0);
+            assertThat(type.compare(normalizeSelect.apply(json), normalizeSelect.apply(json))).isEqualTo(0);
+
+            assertThat(type.compare(normalizeInsert.apply(json), json)).isEqualTo(0);
+            assertThat(type.compare(normalizeInsert.apply(json), normalizeInsert.apply(json))).isEqualTo(0);
```

Comment:
This will pass even without the changes. I also do not really understand how this should trip the issue. The problem is related to casting object literals containing a GeometryCollection vs a Multipoint.

Reply:
Thanks for catching, I reverted this change and added 2 tests in https://github.com/crate/crate/pull/17012/commits/ed2a9dfcd9a30c8a55d6441c34f71345c0f93ae9 (and they fail without new changes). 

One test is specifically for transformation and another to check that implicitCast/sanitizeValue (SELECT/INSERT) actually utilize new logic.

Reply:
I found out that I missed processing `MultiLine` string. I refactored code a bit and improved coverage in https://github.com/crate/crate/pull/17012/commits/6ad56ac442e7eda168d5352896773ed6b6277cc3